subsys/mgmt/mcumgr/lib/mgmt/src/mgmt.c

```diff
@@ -64,30 +73,70 @@ mgmt_register_group(struct mgmt_group *group)
 	sys_slist_append(&mgmt_group_list, &group->node);
 }
 
-void
-mgmt_ntoh_hdr(struct mgmt_hdr *hdr)
+#if defined(CONFIG_MGMT_MCUMGR_MGMT_NOTIFICATION_HOOKS)
+void mgmt_register_callback(struct mgmt_callback *callback)
 {
-	hdr->nh_len = sys_be16_to_cpu(hdr->nh_len);
-	hdr->nh_group = sys_be16_to_cpu(hdr->nh_group);
+	sys_slist_append(&mgmt_callback_list, &callback->node);
 }
 
-void
-mgmt_hton_hdr(struct mgmt_hdr *hdr)
+void mgmt_unregister_callback(struct mgmt_callback *callback)
 {
-	hdr->nh_len = sys_cpu_to_be16(hdr->nh_len);
-	hdr->nh_group = sys_cpu_to_be16(hdr->nh_group);
+	(void)sys_slist_find_and_remove(&mgmt_callback_list, &callback->node);
 }
 
+int mgmt_notify_callbacks(uint32_t event, void *data, size_t data_size)
+{
+	sys_snode_t *snp, *sns;
+	bool failed = false;
+	bool abort_more = false;
+	int rc;
+	int return_rc = MGMT_ERR_EOK;
+	uint16_t group = MGMT_EVT_GET_GROUP(event);
+
+	/*
+	 * Search through the linked list for entries that have registered for this event and
+	 * notify them, the first handler to return an error code will have this error returned
+	 * to the calling function, errors returned by additional handlers will be ignored. If
+	 * all handlers return  Access will be allowed so long as all notification handlers
```

Comment:
```suggestion
	 * all handlers return  Access will be allowed so long as all notification handlers
```
Looks like part of the sentence hasn't been removed after editing.